src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -880,22 +890,79 @@ QgsLegendModel::QgsLegendModel( QgsLayerTree *rootNode, QObject *parent )
 QVariant QgsLegendModel::data( const QModelIndex &index, int role ) const
 {
   // handle custom layer node labels
-  if ( QgsLayerTreeNode *node = index2node( index ) )
+
+  QgsLayerTreeNode *node = index2node( index );
+  QgsLayerTreeModelLegendNode *ltmln = index2legendNode( index ); // Possibly useless
+  QgsLayerTreeLayer *nodeLayer = QgsLayerTree::isLayer( node ) ? QgsLayerTree::toLayer( node ) : nullptr;
+  if ( nodeLayer && ( role == Qt::DisplayRole || role == Qt::EditRole ) )
   {
-    if ( QgsLayerTree::isLayer( node ) && ( role == Qt::DisplayRole || role == Qt::EditRole ) && !node->customProperty( QStringLiteral( "legend/title-label" ) ).isNull() )
+    QString name;
+    QgsSymbolLegendNode *symnode = nullptr;
+    QgsVectorLayer *vlayer = qobject_cast<QgsVectorLayer *>( nodeLayer->layer() );
+    if ( ltmln )
+    {
+      symnode = qobject_cast<QgsSymbolLegendNode *>( ltmln );
```

Comment:
Is this guaranteed to always be a QgsSymbolLegendNode? If not, better do

```
    QgsSymbolLegendNode *symnode = qobject_cast<QgsSymbolLegendNode *>( ltmln );
    if ( symnode )
      name = symnode->symbolLayer();
```

to avoid crashers.

Reply:
done